geoalchemy2/elements.py

```diff
@@ -131,11 +132,38 @@ class WKBElement(_SpatialElement):
 
     Note: you can create ``WKBElement`` objects from Shapely geometries
     using the :func:`geoalchemy2.shape.from_shape` function.
+
+    Extended WKB automatically parse unknown SRID.
+    Warning: unicode data are assumed to be hexadecimal value.
+
+    WKB struct {
+        byte    byteOrder;
+        uint32  wkbType;
+        uint32  SRID;
+        struct  geometry;
+    }
+
+    byteOrder enum {
+        WKB_XDR = 0,  // Most Significant Byte First
+        WKB_NDR = 1,  // Least Significant Byte First
+    }
+
     """
 
     geom_from = 'ST_GeomFromWKB'
     geom_from_extended_version = 'ST_GeomFromEWKB'
 
+    def __init__(self, data, srid=-1, extended=False):
+        if extended and srid == -1:  # unpack srid from WKB struct
+            header = (binascii.unhexlify(data[:18]) if isinstance(data, str_)
+                      else data[:9])  # unicode data assumed to be hex val
```

Comment:
I am picky but I find this form more readable:

```py
if isinstance(data, str_):
    header = binascii.unhexlify(data[:18])
else:
    header = data[:9]
```

Can you please change it? Otherwise I'll do it myself once this is merged :)

Reply:
Of course, no problem.
I chose this way because it is safer to ensure `header` is set before used :wink: 
Otherwise, an alternative is:
```python
header = data[:9]
if isinstance(data, str_):
    header = binascii.unhexlify(data[:18])
```
but the first slicing is useless, in the case of unicode `data`.

Reply:
Relying on `else` is safe too. It is garanteed that `header` will be defined after the `if/else` block. Let's change this and merge! ;-)

Reply:
:+1: